netx/net/sourceforge/jnlp/util/WindowsDesktopEntry.java

```diff
@@ -71,31 +71,36 @@ public void createShortcutOnWindowsDesktop() throws IOException {
     @Override
     public void createWindowsMenu() throws IOException {
         // create menu item
-        // see if menu is defined in jnlp, else don't do it
-        String path = file.getInformation().getShortcut().getMenu().getSubMenu();
-        if (path != null) {
-            path = System.getenv("userprofile") + "/AppData/Roaming/Microsoft/Windows/Start Menu/Programs/" + path;
-            // check to see if menu dir exists and create if not
-            File menuDir = new File(path);
-            if (!menuDir.exists()) {
-                menuDir.mkdir();
-            }
-            String JavaWsBin = XDesktopEntry.getJavaWsBin();
-            String favIcon = getFavIcon();
-            ShellLink sl = ShellLink.createLink(JavaWsBin).setCMDArgs(file.getSourceLocation().toString());
-            // setup uninstall shortcut
-            ShellLink ul = ShellLink.createLink(JavaWsBin).setCMDArgs("-Xclearcache " + file.getFileLocation().toString());
-            if (favIcon != null) {
-                favIcon = favIcon.substring(6);
-                sl.setIconLocation(favIcon);
-                ul.setIconLocation(favIcon);
-            }
-            sl.saveTo(path + "/" + file.getInformation().getTitle()+ ".lnk");
-            ul.saveTo(path + "/Uninstall " + file.getInformation().getTitle() + ".lnk");
-            // write shortcuts to list
-            manageShortcutList(ManageMode.A, path + "/" + file.getInformation().getTitle() + ".lnk");
-            manageShortcutList(ManageMode.A, path + "/Uninstall " + file.getInformation().getTitle() + ".lnk");
+        String path;
+        try {
+            path = file.getInformation().getShortcut().getMenu().getSubMenu();
+        }
+        catch (NullPointerException npe) {
+            path = null;
+        }        
+        if (path == null) {
+            path = XDesktopEntry.getDesktopIconName(file);
+        }
+                        
+        path = System.getenv("userprofile") + "/AppData/Roaming/Microsoft/Windows/Start Menu/Programs/" + path;
```

Comment:
if you set the path here alll line above do not make sense since at the end it will be overridden here...

Reply:
That's not true. The previously determined path gets only a prefix.

Reply:
yes, missed the `+path` ;)
But in that case the line should be:
```suggestion
        final String realPath = System.getenv("userprofile") + "/AppData/Roaming/Microsoft/Windows/Start Menu/Programs/" + path;
```
or something. Maybe rename the first usage of `path` to `pathPrefix`. Main goal should be to have final variables wherever possible to minimise mutability.